### ISSUE TITLE
Group results on Search need title element/label cleanup + other Group accessibility fixes for descriptions

### ISSUE THREAD

**Issue by manfromjupyter** [nbgallery/nbgallery#530]
Searching for a group and then hovering over it needs to be cleaned up in terms of how it appears, needs to describe it also to screenreaders (currently just tells the user "link"). Lastly needs to be prepared for groups where there is no description. Currently you may see. "16 notebooks = {Twilight Fan Club description}".

I think groups should just be changed to description on hover on search page (no mention of their total notebooks). And add aria-label equal to `"{group name} with description {group_description}"` if description is not blank, else have the aria label say `"{group_name} with no description."`

In addition, this same aria-label should be applied to the /groups and /users/{ID}/groups pages as well when groups are being hovered over

![group on search pages](https://user-images.githubusercontent.com/51969207/113320639-b744c880-92e0-11eb-892c-893ab54254a8.PNG)
